census/handler.go

```diff
@@ -315,13 +314,20 @@ func (m *Manager) Handler(ctx context.Context, r *api.MetaRequest, isAuth bool,
 		}
 		var err error
 		if r.Method == "dump" {
-			resp.CensusDump, err = tr.Dump(root)
-		} else {
-			var vals [][]byte
-			resp.CensusKeys, vals, err = tr.DumpPlain(root)
-			for _, v := range vals {
-				resp.CensusValues = append(resp.CensusValues, types.HexBytes(v))
+			snapshot, err := tr.FromRoot(root)
+			if err != nil {
+				resp.SetError(err)
 			}
+			resp.CensusDump, err = snapshot.Dump()
+		} else {
+			// TODO the method DumpPlain no longer exists, clarify
+			// the difference of usage between 'dump' & 'dumpPlain'.
+
+			// var vals [][]byte
+			// resp.CensusKeys, vals, err = tr.DumpPlain(root)
+			// for _, v := range vals {
+			//         resp.CensusValues = append(resp.CensusValues, types.HexBytes(v))
+			// }
```

Comment:
@p4u Which was the purpose of having `Dump` & `DumpPlain`? Could the `DumpPlain` be removed?